### PR TITLE
Extend :load REPL command to accept directories (BT-784)

### PR DESCRIPTION
**Pull request by jamesc**
## Summary

- `:load <directory>` recursively discovers and loads all `.bt` files, sorted by full path for deterministic order
- Compilation errors in individual files don't stop loading remaining files; summary shows successes and failures
- `:reload` after a directory load re-walks the directory, picking up new files added since the original load

## Changes

- **`client.rs`**: Replace `last_loaded_file: Option<String>` with `LastLoadedPath` enum tracking file vs directory
- **`mod.rs`**: Add `discover_bt_files` recursive walker and `load_directory` handler; update `:load` and `:reload` command routing
- **`display.rs`**: Update help text to mention directory support
- **`e2e.rs`**: Add directory loading, `:reload`, and `:l` shorthand support to the e2e test framework
- **New fixtures**: `tests/e2e/fixtures/load_dir_test/{alpha,beta}.bt` for testing
- **New e2e test**: `tests/e2e/cases/load_directory.bt` covering directory load, reload, shorthand, and errors

## Test plan

- [x] Unit tests for `discover_bt_files` (sorted order, recursive, hidden dirs, empty, nonexistent)
- [x] E2E tests for `:load <dir>`, `:reload` after dir load, `:l` shorthand, single file after dir
- [x] `just ci` passes (all 595 e2e tests pass)

[Linear: BT-784](https://linear.app/beamtalk/issue/BT-784)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Load entire directories with :load <directory> (recursively) and get per-file success/failure summaries; :l is an alias.
  * :reload now targets the last-loaded file or directory and also accepts an explicit path; :r is an alias.
  * New REPL commands: :actors, :kill, :inspect, :sessions, and :info; :info shows symbol details.

* **Documentation**
  * Updated REPL help text to describe directory loading, reload semantics, shorthands, and new commands.

* **Tests**
  * Added end-to-end tests and fixtures for directory load, reload behavior, and edge cases.

### CODE REVIEW

File: tests/e2e/cases/load_directory.bt [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+b getValue
+// => 2
+
+// ===========================================================================
+// :reload — RELOADS THE DIRECTORY (re-walks)
+// ===========================================================================
+
+:reload
+// => Reloaded 2 files from tests/e2e/fixtures/load_dir_test
```
[the remark this answers is not in the document]
Comment (jamesc) [reply tests/e2e/cases/load_directory.bt]: Thanks — updated the test expectation to include the trailing slash and fixed the ERROR assertion; changes committed in 2cccf430d0bedd6e55db83d3efc95e16bd2f0469.

[the remark this answers is not in the document]
Comment (jamesc) [reply tests/e2e/cases/load_directory.bt]: Thanks — updated the test expectation and ERROR assertion in commits 2cccf430d0bedd6e55db83d3efc95e16bd2f0469 and 1af02db9af8de77c078e079dcba339c1c83596d6; pushed to branch worktree-BT-784.

File: crates/beamtalk-cli/src/commands/repl/mod.rs [hunk 2]
```diff
@@ -1194,6 +1215,118 @@ pub(crate) fn strip_path_quotes(arg: &str) -> &str {
...
+                } else {
+                    loaded += 1;
+                }
+            }
+            Err(e) => {
+                let file_name = file
+                    .file_name()
+                    .map_or_else(|| file_str.to_string(), |n| n.to_string_lossy().to_string());
+                eprintln!("  Error loading {file_name}: {e}");
+                failed.push(file_name);
```
[the remark this answers is not in the document]
Comment (jamesc) [reply crates/beamtalk-cli/src/commands/repl/mod.rs]: Thanks — good suggestion. Noted for a follow-up; will open a small follow-up PR to report relative paths in directory load error reporting.

File: tests/e2e/cases/load_directory.bt [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+// ===========================================================================
+
+// Single file load should still work
+:load tests/e2e/fixtures/point.bt
+// => Loaded: Point
+
+p := Point new
+// => _
+
+p getX
```
[the remark this answers is not in the document]
Comment (jamesc) [reply tests/e2e/cases/load_directory.bt]: Thanks — good suggestion to add // load directives for fixtures; will add these in a follow-up change.

File: crates/beamtalk-cli/src/commands/repl/display.rs [hunk 4]
```diff
@@ -190,3 +188,63 @@ pub(crate) fn display_codegen(core_erlang: &str) {
...
+        .and_then(serde_json::Value::as_bool)
+        .unwrap_or(false);
+    let symbol = info
+        .get("symbol")
+        .and_then(serde_json::Value::as_str)
+        .unwrap_or("?");
+
+    if !found {
+        println!("Symbol not found: {symbol}");
+        return;
```
[the remark this answers is not in the document]
Comment (jamesc) [reply crates/beamtalk-cli/src/commands/repl/display.rs]: Thanks — noted. Routing this diagnostic to stderr is a good idea; will address in a follow-up PR.
